main.go

```diff
@@ -105,10 +108,10 @@ func emulatePing() ping {
 	}
 }
 
-func getPin(pin int) rpio.Pin {
-	thePin := rpio.Pin(pin)
+func getPin(pin string) gpio.PinIO {
+	thePin := gpioreg.ByName(pin)
 
-	if err := rpio.Open(); err != nil {
+	if err := thePin.In(gpio.PullUp, gpio.FallingEdge); err != nil {
```

Comment:
did you confirm the `gpioPing` isOpen state is correct and not inverted? By pulling this up you may have inverted it.
Also im not sure FallingEdge is required here?